src/__tests__/PaletteProvider.test.js

```diff
@@ -102,33 +102,24 @@ describe('PaletteProvider', () => {
     );
   });
 
-  it('should throw error if `onError` was not passed and palette creation fails', () =>
-    new Promise(resolve => {
-      function checkErrorAndFinish(error) {
-        expect(error.message).toMatch(/MaterialPaletteProvider.*test/);
-        resolve();
-      }
-
-      createMaterialPalette.mockImplementation(() => ({
-        then() {
-          return this;
-        },
-        catch(errorHandler) {
-          try {
-            errorHandler(new Error('test'));
-          } catch (error) {
-            checkErrorAndFinish(error);
-          }
-        },
-      }));
-
-      render(
-        <PaletteProvider image={0} options={{ type: 'vibrant' }}>
-          <Text>Test</Text>
-        </PaletteProvider>,
-      );
+  it('should throw error if `onError` was not passed and palette creation fails', () => {
+    createMaterialPalette.mockImplementation(() => ({
+      then(success, failure) {
+        try {
+          failure(new Error('test'));
+        } catch (error) {
+          expect(error.message).toMatch(/MaterialPaletteProvider.*test/);
```

Comment:
Why not `expect(error.message).toBe('Uncaught MaterialPaletteProvider exception: test')` for more accuracy of what's being thrown?